### PR TITLE
irmin-pack: move Store.Atomic_write into its own module

### PR DESCRIPTION
**Pull request by samoht**
On top of #1377

### CODE REVIEW

File: src/irmin-pack/store_intf.ml [hunk 1]
```diff
@@ -91,7 +91,4 @@ module type Sigs = sig
   module type Specific = Specific
   module type S = S
   module type Maker = Maker
-
```
Comment (Ngoguey42) [inline src/irmin-pack/store_intf.ml:94]: `store.ml` is now empty. I suppose you are about to delete the file.

File: src/irmin-pack/layered/layered_store.ml [hunk 2]
```diff
@@ -333,8 +333,10 @@ end
 
 module Atomic_write
```
Comment (Ngoguey42) [inline src/irmin-pack/layered/layered_store.ml:334]: This `Atomic_write` could be moved to his own file too. Maybe in a later PR

**Review by Ngoguey42 — APPROVED** [review]

File: src/irmin-pack/store_intf.ml (hunk 1 above)
Comment (samoht) [reply src/irmin-pack/store_intf.ml:94]: Indeed, see #1379  :-)

File: src/irmin-pack/layered/layered_store.ml (hunk 2 above)
Comment (samoht) [reply src/irmin-pack/layered/layered_store.ml:334]: good idea, I've added it to https://github.com/mirage/irmin/issues/1340

**Pull request merged**